src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const currentArrayLength = this.length;
+    const joinSeparator = separator + '';
```

Comment:
I propose to use a more readable string transform for it.

```suggestion
    const joinSeparator = String(separator);
```